remotes/mocks_test.go

```diff
@@ -90,3 +99,71 @@ func (f *mockFetcher) Fetch(ctx context.Context, desc ocischemav1.Descriptor) (i
 	f.indexBuffers = f.indexBuffers[1:]
 	return rc, nil
 }
+
+type mockReadCloser struct {
+}
+
+func (rc mockReadCloser) Read(p []byte) (n int, err error) {
+	return 0, io.EOF
+}
+
+func (rc mockReadCloser) Close() error {
+	return nil
+}
+
+type mockImageAPIClient struct {
+}
+
+func newMockImageAPIClient() *mockImageAPIClient {
```

Comment:
I think this mock shows we need our own interface here. We should add a simple interface with only what we use:
```go
type ImageClient interface{
    ImagePush(ctx context.Context, ref string, options types.ImagePushOptions) (io.ReadCloser, error)
}
```
Using that simple interface means a simpler mock 😸

Reply:
Good idea.
Done.